refinery/analysis_manager/models.py

```diff
@@ -45,9 +49,19 @@ def set_galaxy_history_state(self, state):
     def refinery_import_state(self):
         return get_task_group_state(self.refinery_import_task_group_id)
 
-    def galaxy_import_state(self):
+    def galaxy_file_import_state(self):
         return get_task_group_state(self.galaxy_import_task_group_id)
 
+    def tool_based_galaxy_file_import_state(self):
+        if self.galaxy_history_state and self.galaxy_import_progress != 0:
+            galaxy_file_import_state = [{
+                'state': self.galaxy_history_state,
+                'percent_done': self.galaxy_import_progress
+            }]
+        else:
+            galaxy_file_import_state = []
```

Comment:
Is the if-then necessary? It feels like a case where we're introducing a variation in the structure and all the code downstream will need to take both possibilities into account.